framework/src/engine/legacy/legacy_chain_handler.ts

```diff
@@ -54,16 +54,50 @@ export class LegacyChainHandler {
 
 		for (const bracket of this._legacyConfig.brackets) {
 			try {
-				await this._storage.getLegacyChainBracketInfo(Buffer.from(bracket.snapshotBlockID, 'hex'));
+				const encodedBracketInfo = await this._storage.getLegacyChainBracketInfo(
+					Buffer.from(bracket.snapshotBlockID, 'hex'),
+				);
+
+				const decodedBracketInfo = codec.decode<LegacyChainBracketInfo>(
+					legacyChainBracketInfoSchema,
+					encodedBracketInfo,
+				);
+
+				// Update the existing legacyInfo
+				let startBlock;
+				try {
+					startBlock = await this._storage.getBlockByHeight(bracket.startHeight);
+				} catch (error) {
+					if (!(error instanceof NotFoundError)) {
+						throw error;
+					}
+				}
+
+				await this._storage.setLegacyChainBracketInfo(Buffer.from(bracket.snapshotBlockID, 'hex'), {
+					...decodedBracketInfo,
+					startHeight: bracket.startHeight,
+					snapshotBlockHeight: bracket.snapshotHeight,
+					// if start block already exists then assign to lastBlockHeight
+					lastBlockHeight: startBlock ? bracket.startHeight : bracket.snapshotHeight,
+				});
 			} catch (err) {
 				if (!(err instanceof NotFoundError)) {
 					throw err;
 				}
 				// Save config brackets in advance, these will be used in next step (`sync`)
-				await this._storage.setLegacyChainBracketInfo(Buffer.from(bracket.snapshotBlockID), {
+				let startBlock;
+				try {
+					startBlock = await this._storage.getBlockByHeight(bracket.startHeight);
+				} catch (error) {
+					if (!(error instanceof NotFoundError)) {
+						throw error;
+					}
+				}
+				await this._storage.setLegacyChainBracketInfo(Buffer.from(bracket.snapshotBlockID, 'hex'), {
```

Comment:
maybe this can be cleaned up with
```
try {
	const bracketInfoExist = await this._storage.legacyChainBracketInfoExist(...);
	if (!bracketInfoExist) {
		await this._storage.setLegacyChainBracketInfo(Buffer.from(bracket.snapshotBlockID, 'hex'), {
			startHeight: bracket.startHeight,
			snapshotBlockHeight: bracket.snapshotHeight,
			lastBlockHeight: bracket.snapshotHeight,
		});
		continue;
	}

	const encodedBracketInfo = await this._storage.getLegacyChainBracketInfo(
		Buffer.from(bracket.snapshotBlockID, 'hex'),
	);
	const decodedBracketInfo = codec.decode < LegacyChainBracketInfo > (
		legacyChainBracketInfoSchema,
		encodedBracketInfo,
);
	const startBlock = await this._storage.getBlockByHeight(bracket.startHeight);

	const bracketInfo = await this._storage.setLegacyChainBracketInfo(
		...decodedBracketInfo,
		startHeight: bracket.startHeight,
		snapshotBlockHeight: bracket.snapshotHeight,
		// if start block already exists then assign to lastBlockHeight
		lastBlockHeight: startBlock ? bracket.startHeight : bracket.snapshotHeight,
	);
} catch (error) {
	// NotFoundError is expected for this._storage.getBlockByHeight, and if it doesn't exist, no need to update
	if (!(error instanceof NotFoundError)) {
		throw error;
	}	
}
```